gtest/src/system.rs

```diff
@@ -174,13 +174,69 @@ impl System {
             .try_init();
     }
 
-    /// Send raw message dispatch.
-    pub fn send_dispatch(&self, dispatch: Dispatch) -> RunResult {
-        self.0.borrow_mut().validate_and_run_dispatch(dispatch)
+    /// Run next block.
+    ///
+    /// Block execution model is the following:
+    /// - increase the block number, update the timestamp
+    /// - process tasks from the task pool
+    /// - process messages in the queue.
+    ///
+    /// The system is always initialized with a 0 block number. Current block
+    /// number in the system is the number of the already executed block,
+    /// therefore block execution starts with a block info update (block
+    /// number increase, timestamp update). For example, if current block
+    /// number is 2, it means that messages and tasks on 2 were executed, so
+    /// the method goes to block number 3 and executes tasks and messages for
+    /// the updated block number.
+    ///
+    /// Task processing basically tries to execute the scheduled to the specific
+    /// block tasks:
+    /// - delayed sending
+    /// - waking message
+    /// - removing from the mailbox
+    /// - removing reservations
+    /// - removing stalled wait message.
+    ///
+    /// Messages processing executes messages until either queue becomes empty
+    /// or block gas allowance is fully consumed.
+    pub fn run_next_block(&self) -> BlockRunResult {
```

Comment:
Should it be named run_to_next_block? Or maybe simplified to run_block?

From an API standpoint, this naming seems a bit odd. If a function is named `run_next_block`, what happens to the current one? Where is the `run_current_block function`, etc.?

Reply:
discussed